osf/models/preprint_service.py

```diff
@@ -205,11 +210,70 @@ def save(self, *args, **kwargs):
         saved_fields = self.get_dirty_fields() or []
         old_subjects = kwargs.pop('old_subjects', [])
         ret = super(PreprintService, self).save(*args, **kwargs)
-
         if (not first_save and 'is_published' in saved_fields) or self.is_published:
             enqueue_postcommit_task(on_preprint_updated, (self._id,), {'old_subjects': old_subjects}, celery=True)
+
+        request, user_id = get_request_and_user_id()
+        request_headers = {}
+        if not isinstance(request, DummyRequest):
+            request_headers = {
+                k: v
+                for k, v in get_headers_from_request(request).items()
+                if isinstance(v, basestring)
+            }
+        if saved_fields and self.check_spam(self.node.creator, saved_fields, request_headers):
+            super(PreprintService, self).save()
         return ret
 
+    def _get_spam_content(self, saved_fields):
+        spam_fields = self.node.SPAM_CHECK_FIELDS if self.node.is_public and 'is_public' in saved_fields else self.node.SPAM_CHECK_FIELDS.intersection(
```

Comment:
This looks to be much the same logic as what's in `AbstractNode._get_spam_content`. However, `saved_fields` will contain the saved preprint fields, which should have approximately no intersection with `AbstractNode.SPAM_CHECK_FIELDS`, and `PreprintService.is_published` is roughly equivalent to `AbstractNode.is_public`.

It should be safe to assume that when a node is created or updated for a preprint, that node will be checked for spam. Rather than trying to check both the preprint and node for spam when the preprint is created / updated, it might make more sense to simply update the node spam handling to flag any preprints as potential spam if the node smells spammy, at least until preprints are actually a first-class object. At that point (after the preprint/node divorce, when preprint content is stored on the preprint), we can add `SPAM_CHECK_FIELDS` to `PreprintService` and actually serialize and send that content to Akismet.